ratis-server/src/main/java/org/apache/ratis/server/RaftServer.java

```diff
@@ -47,6 +47,8 @@
   /** @return the groups the server is part of. */
   Iterable<RaftGroup> getGroups() throws IOException;
 
+  StateMachine getStateMachine(RaftGroupId groupId) throws IOException;
```

Comment:
I wasn't make this change in RATIS-1084 because I wasn't sure the consequence to add such function to `RaftServer` interface as `RaftServer` interface looks very important. Thus I decided to defer this change to another change.

But based on this PR, making change to `RaftServer` is not that hard.